src/crawlers/puppeteer_crawler.js

```diff
@@ -182,8 +176,9 @@ class PuppeteerCrawler {
     constructor(options) {
         const {
             handlePageFunction,
-            gotoFunction,
-            handlePageTimeoutSecs,
+            gotoFunction = this._defaultGotoFunction,
```

Comment:
Will it work without binding? That function contains `{ timeout: this.gotoTimeoutMillis }` i.e., `this`

This won't fail that function but timeout will be `undefined`.

Reply:
It's saved and invoked as `this.gotoFunction` so it will.